src/trg-main-window.c

```diff
@@ -34,6 +34,8 @@
 #include <curl/curl.h>
 #if HAVE_LIBAPPINDICATOR
 #include <libappindicator/app-indicator.h>
+#elif HAVE_LIBAYATANA_APPINDICATOR
+#include <libayatana-appindicator/app-indicator.h>
```

Comment:
Much better than what I was thinking, but are these APIs confirmed 1:1? If libappindicator is deprecated should it really be the default?

Reply:
I wanted the patch to be minimally intrusive, which is why it yanks in libappindicator first. I didn't try to prevent it from linking in both; someone should probably look at that, perhaps by ensuring that both are not simultaneously enabled. It's hard for me to test if that works, since I'm running Debian and don't have libappindicator-dev to kick around.

Just did a quick check for the situation upstream with these libraries. The deal is that libappindicator is Ubuntu-only by virtue of using ubuntu-only facilities, and libayatana-appindicator3 is a fork, intended to be plug-compatible, that uses only generic not-ubuntu-only facilities, but still works on Ubuntu. So it's not totally unreasonable to support both, at least until libappindicator dies, should that ever happen.

The larger question of why this is being done in a busy-work-for-everyone fashion ... that, I do not know the answer to.

Reply:
Hmm, I didn't even think about linking both at once, that could definitely be a bad idea. If you did something like
```meson
trg_deps += libappindicator_dep ? libappindicator_dep.found() : libayatana_appindicator_dep
```
Maybe that would be enough?

Reply:
That logic seems reasonable, but it might be more robust to use a similar conditional to ensure that at most one of libappindicator and libataya-appindicator3 is enabled, and have some other flag (`HAVE_LIBSOMETHINGAPPINDICATOR`) which is turned on if either is on. This might allow the C code to be a bit cleaner for shared code that works with either library.

Reply:
Yes we want to use only one at a time.

>  Just did a quick check for the situation upstream with these libraries. The deal is that libappindicator is Ubuntu-only by virtue of using ubuntu-only facilities, and libayatana-appindicator3 is a fork, intended to be plug-compatible, that uses only generic not-ubuntu-only facilities, but still works on Ubuntu. 

This is also odd phrasing. At this point only Debian ships libayatana and nobody else does because its an API break. libappindicator is packaged everywhere, while not well maintained.

Reply:
> it might be more robust to use a similar conditional to ensure that at most one of libappindicator and libataya-appindicator3 is enabled, and have some other flag (`HAVE_LIBSOMETHINGAPPINDICATOR`) which is turned on if either is on.

I don't see how this could work if the `#include`s are different. If the functions we use are API compatible then at most doing 
```meson 
conf.set10('HAVE_APPINDICATOR', libappindicator_dep or libayatana_appindicator_dep)
```
would be all you need, and then replacing your `HAVE_LIBAPPINDICATOR || HAVE_LIBAYATANA` lines with that. You're still going to need to have the `#include` line be separate though.

Reply:
I'm not versed in the politics of application indication, so all I really know is that other programs (gromit-mpx, to take an example I'm familiar with) have added support for libayatana-appindicator3 and fall back to libappindicator only if unavailable, and there's some kind of split for some reason I don't understand having something to do with Unity.

Re conditionals, I don't think there's any disagreement. You need to be able to distinguish between the two libraries using different HAVE_'s, for different #includes if nothing else. Maybe it would be more modular to have a single one that's on if either's available, because they're supposed to be plug compatible after all.